### ISSUE TITLE
[OSX] Possible "semi-zombie state" caused by conflict with Zsh?

### ISSUE THREAD

**Issue by jmurinello** [tmux/tmux#500]
- [tmux session in weird semi-zombie state](https://github.com/tmux/tmux/issues/298)
- Zsh version: zsh 5.2 (x86_64-apple-darwin15.3.0)
- tmux 2.2

In reference to the issue linked above (were I've just commented), in my specific case it was a Zsh configuration that was causing the _"semi-zombie"_ state on `tmux`.

Do you guys have any clue on why having `unsetopt HUP` in the `.zshrc` file causes `tmux` to behave like that? Because otherwise `tmux` attach the session just fine.

I opened an issue ticket because the `unsetopt HUP` option was by default added by the `Zsh framework` I'm using and that could be the case of many others.

Cheers!

**Comment by nicm** [comment]
No idea, what does it do?

**Comment by jmurinello** [comment]
Zsh option `HUP` is disabled

> to prevent SIGHUP to jobs on shell close

**Comment by nicm** [comment]
Does this still happen in Git master or not?

**Comment by jmurinello** [comment]
Yes.

**Comment by nicm** [comment]
What exactly is happening? This thread makes no sense.

**Comment by nicm** [comment]
The other thread.

**Comment by jmurinello** [comment]
Ok, so what was happening with me was the following:

Start a new tmux session, quit the terminal with the session still attached, open the terminal and attach tmux session...

End result: the `shell` stopped rendering whatever I was inputting.

I was still able to preform `tmux` commands, such as opening a new `tmux window` in the same session and switch back and forward between windows, so for instance:

In window 1 typed  `ls + enter` without seeing what I was typing nor the result of that command, switch to window 2 and switch back to window 1 and there it was my $HOME directory ls.

Because I could preform `tmux`commands I then `tmux kill-server` and the shell behaved naturally outside `tmux` and I could create a new `tmux session` and work normally.

This type of behaviour is reported by many in the mentioned thread and it seems that its related with the killing of the terminal without detaching the session and upon recovering that session.

The bottom line is:
If you add `unsetopt HUP` (which is related with the shell closing) in the your zsh configuration file and you can't replicate this "zombie state"behaviour, then I was an isolated case. 

Sorry for the long post.
Cheers!

**Comment by nicm** [comment]
Do you do the unsetopt command outside tmux or inside?

**Comment by nicm** [comment]
If the tmux client doesn't get the HUP when it is closed then it will never exit, with 2.1 and 2.2 this will mean it will eventually stop accepting data which will mean tmux will stop reading from panes. But in master this shouldn't happen. Please run

tmux -Ltest -f/dev/null -vvvv new

then reproduce, kill tmux and send me the tmux-server-*.log file.

**Comment by jmurinello** [comment]
I'm sorry for my ignorance but am I supposed to reproduce using the session created by the command above?, because I don't know how to attach that session. It says something like "no session running on private server".

But yet, if I create another session with the same command and `<prefix> + s ` I can see both sessions and both attached.

I'm really sorry.

Anyway, manage to reproduce without using `-Ltest` and using only `tmux -f/dev/null -vvvv new`

Here's the log:

[tmux-server-3012.log](https://gist.github.com/jmurinello/1577d70f47134c04fc6ed2e0cadf706c)

**Comment by nicm** [comment]
This log is not from a tmux built from Git master.

**Comment by jmurinello** [comment]
... damn. I though brew install tmux would get it from Git master

I brew uninstall tmux and installed tmux via git followed the README file instructions.

Everything is fine. No freezing no nothing even with `unsetopt HUP`.

Guess that's a big thank you.

**Comment by krader1961** [comment]
@jmurinello, Just an FYI for future use: You can force HomeBrew to use the latest github source for a project by running `brew install --devel` or `brew install -HEAD`.

**Comment by codeHatcher** [comment]
`brew upgrade tmux -HEAD` didn't work for me as a FYI

**Comment by codeHatcher** [comment]
oh, I see, I had to uninstall, then `brew install tmux -HEAD` to get from V2.1 to V2.2
